remove `|| true` hacks from `make nix-purge`

### Problem

I introduced a bunch of hacks in the purge script for Darwin here : https://github.com/status-im/status-mobile/pull/18192
We should remove these hacks and modify the script to work without omitting failures

Removed hacks here : https://github.com/status-im/status-mobile/commit/1e2acbfacba9b5d78c596cbb90e5f6b47ec5e5ea

first failure I get when i do `make nix-purge` is this : 

```
make nix-purge
nix/scripts/purge.sh
Are you sure you want to purge Nix?
[y/n]: y
Detected multi-user Nix installation.
Password:
make: *** [nix-purge] Error 3
```

I retry `make nix-purge` and this time I get : 

```
make nix-purge
nix/scripts/purge.sh
Are you sure you want to purge Nix?
[y/n]: y
Detected multi-user Nix installation.
Password:
Unload failed: 5: Input/output error
Try running `launchctl bootout` as root for richer errors.
make: *** [nix-purge] Error 3
```

--- a/nix/scripts/kill_proc_prompt.sh
+++ b/nix/scripts/kill_proc_prompt.sh
@@ -1,16 +1,14 @@
 #!/usr/bin/env bash
+set -Eeuo pipefail
+# script to prompt user for terminating a process
 
-# Function to prompt user for terminating a process
-kill_proc_prompt() {
-    local pid=$1
+pid="${1}"
 
-    # Ask the user whether to terminate the process
-    read -p "Do you want to terminate this process? (y/n): " choice
-    if [[ $choice == "y" ]]; then
-        sudo kill $pid
-        echo "Process $pid terminated."
-    else
-        echo "Process not terminated. Please close it manually and retry."
-        return 1
-    fi
-}
+read -p "Do you want to terminate this process? (y/n): " choice
+if [[ "${choice}" == "y" ]]; then
+    sudo kill "${pid}"
+    echo "Process ${pid} terminated."
+else
+    echo "Process not terminated. Please close it manually and retry."
+    exit 1
+fi

--- a/nix/scripts/kill_proc_prompt.sh
+++ b/nix/scripts/kill_proc_prompt.sh
@@ -1,0 +1,16 @@
+#!/usr/bin/env bash
+
+# Function to prompt user for terminating a process
+kill_proc_prompt() {
+    local pid=$1
+
+    # Ask the user whether to terminate the process
+    read -p "Do you want to terminate this process? (y/n): " choice
+    if [[ $choice == "y" ]]; then
+        sudo kill $pid
+        echo "Process $pid terminated."
+    else
+        echo "Process not terminated. Please close it manually and retry."
+        return 1
+    fi
+}

--- a/nix/scripts/purge.sh
+++ b/nix/scripts/purge.sh
@@ -4,6 +4,7 @@
 GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
 source "${GIT_ROOT}/nix/scripts/lib.sh"
 source "${GIT_ROOT}/scripts/colors.sh"
+source "${GIT_ROOT}/nix/scripts/kill_proc_prompt.sh"
 
 nix_purge_linux_multi_user_service() {
     NIX_SERVICES=(nix-daemon.service nix-daemon.socket)
@@ -22,12 +23,10 @@ nix_purge_linux_multi_user_users() {
 }
 
 nix_purge_darwin_multi_user_service() {
-    cd /Library/LaunchDaemons
-    NIX_SERVICES=(org.nixos.darwin-store.plist org.nixos.nix-daemon.plist)
-    for NIX_SERVICE in "${NIX_SERVICES[@]}"; do
-        sudo launchctl unload "${NIX_SERVICE}"
-        sudo launchctl remove "${NIX_SERVICE}"
-    done
+    sudo launchctl unload /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+    sudo rm /Library/LaunchDaemons/org.nixos.nix-daemon.plist
+    sudo launchctl unload /Library/LaunchDaemons/org.nixos.darwin-store.plist
+    sudo rm /Library/LaunchDaemons/org.nixos.darwin-store.plist
 }
 
 nix_purge_darwin_multi_user_users() {
@@ -50,16 +49,8 @@ nix_purge_darwin_multi_user_volumes() {
         local pid=$(lsof +D /nix | awk 'NR==2{print $2}')
         if [ -n "$pid" ]; then
             echo "The volume /nix is in use by process ID $pid."
+            kill_proc_prompt "$pid" || return 1
 
-            # Ask the user whether to terminate the process
-            read -p "Do you want to terminate this process? (y/n): " choice
-            if [[ $choice == "y" ]]; then
-                sudo kill $pid
-                echo "Process $pid terminated."
-            else
-                echo "Process not terminated. Please close it manually and retry."
-                return 1
-            fi
         else
             echo "No process found using /nix. Manual intervention required."
             return 1
@@ -117,7 +108,7 @@ if (return 0 2>/dev/null); then
     return
 fi
 
-# Confirm user decission, unless --force is used.
+# Confirm user decision, unless --force is used.
 if [[ "${1}" != "--force" ]]; then
     echo -e "${YLW}Are you sure you want to purge Nix?${RST}" >&2
     read -p "[y/n]: " -n 1 -r
@@ -129,7 +120,7 @@ if [[ "${1}" != "--force" ]]; then
 fi
 
 NIX_INSTALL_TYPE=$(nix_install_type)
-# Purging /nix on NixOS would be disasterous.
+# Purging /nix on NixOS would be disastrous.
 if [[ "${NIX_INSTALL_TYPE}" == "nixos" ]]; then
     echo -e "${RED}You should not purge Nix files on NixOS!${RST}" >&2
     exit

--- a/nix/scripts/purge.sh
+++ b/nix/scripts/purge.sh
@@ -4,7 +4,6 @@
 GIT_ROOT=$(cd "${BASH_SOURCE%/*}" && git rev-parse --show-toplevel)
 source "${GIT_ROOT}/nix/scripts/lib.sh"
 source "${GIT_ROOT}/scripts/colors.sh"
-source "${GIT_ROOT}/nix/scripts/kill_proc_prompt.sh"
 
 nix_purge_linux_multi_user_service() {
     NIX_SERVICES=(nix-daemon.service nix-daemon.socket)
@@ -47,9 +46,9 @@ nix_purge_darwin_multi_user_volumes() {
 
         # Identify the process using the volume
         local pid=$(lsof +D /nix | awk 'NR==2{print $2}')
-        if [ -n "$pid" ]; then
+        if [[ -n "$pid" ]]; then
             echo "The volume /nix is in use by process ID $pid."
-            kill_proc_prompt "$pid" || return 1
+            "${GIT_ROOT}/nix/scripts/kill_proc_prompt.sh" "${pid}" || return 1
 
         else
             echo "No process found using /nix. Manual intervention required."

--- a/nix/scripts/purge.sh
+++ b/nix/scripts/purge.sh
@@ -25,8 +25,8 @@ nix_purge_darwin_multi_user_service() {
     cd /Library/LaunchDaemons
     NIX_SERVICES=(org.nixos.darwin-store.plist org.nixos.nix-daemon.plist)
     for NIX_SERVICE in "${NIX_SERVICES[@]}"; do
-        sudo launchctl unload "${NIX_SERVICE}" || true
-        sudo launchctl remove "${NIX_SERVICE}" || true
+        sudo launchctl unload "${NIX_SERVICE}"
+        sudo launchctl remove "${NIX_SERVICE}"
     done
 }
 
@@ -71,9 +71,9 @@ nix_purge_darwin_multi_user_volumes() {
 
 nix_purge_multi_user() {
     if [[ $(uname -s) == "Darwin" ]]; then
-        nix_purge_darwin_multi_user_service || true
-        nix_purge_darwin_multi_user_users || true
-        nix_purge_darwin_multi_user_volumes || true
+        nix_purge_darwin_multi_user_service
+        nix_purge_darwin_multi_user_users
+        nix_purge_darwin_multi_user_volumes
     else
         nix_purge_linux_multi_user_service
         nix_purge_linux_multi_user_users

